app/webpacker/components/Panel/pages/FixResultsPage/index.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Button, Form, Header, Loader,
+} from 'semantic-ui-react';
+import { useQuery } from '@tanstack/react-query';
+import { IdWcaSearch } from '../../../SearchWidget/WcaSearch';
+import { events, roundTypes } from '../../../../lib/wca-data.js.erb';
+import { editResultUrl } from '../../../../lib/requests/routes.js.erb';
+import SEARCH_MODELS from '../../../SearchWidget/SearchModel';
+import getCompetitions from './api/getCompetitions';
+import Errored from '../../../Requests/Errored';
+import getEvents from './api/getEvents';
+import getResults from './api/getResults';
+
+function FixResultsPage() {
+  const [formValues, setFormValues] = useState({});
+
+  const {
+    wcaId, competitionId, eventId, resultId,
+  } = formValues;
+
+  const {
+    data: competitionsList,
+    isFetching: competitionsFetching,
+    isError: competitionsError,
+  } = useQuery({
+    queryKey: ['fix-results-competitions', wcaId],
+    queryFn: () => getCompetitions({ wcaId }),
+    enabled: !!wcaId,
+  });
+
+  const {
+    data: eventsList,
+    isFetching: eventsFetching,
+    isError: eventsError,
+  } = useQuery({
+    queryKey: ['fix-results-events', wcaId, competitionId],
+    queryFn: () => getEvents({ wcaId, competitionId }),
+    enabled: !!competitionId,
+  });
+
+  const {
+    data: resultsList,
+    isFetching: resultsFetching,
+    isError: resultsError,
+  } = useQuery({
+    queryKey: ['fix-results-rounds', wcaId, competitionId, eventId],
+    queryFn: () => getResults({ wcaId, competitionId, eventId }),
+    enabled: !!eventId,
+  });
+
+  useEffect(() => setFormValues((prevFormValues) => {
+    if (prevFormValues.wcaId !== formValues.wcaId) {
+      return {
+        ...prevFormValues,
+        competitionId: undefined,
+        eventId: undefined,
+        resultId: undefined,
+      };
+    }
+    if (prevFormValues.competitionId !== formValues.competitionId) {
+      return {
+        ...prevFormValues,
+        eventId: undefined,
+        resultId: undefined,
+      };
+    }
+    if (prevFormValues.eventId !== formValues.eventId) {
+      return {
+        ...prevFormValues,
+        resultId: undefined,
+      };
+    }
+    return prevFormValues;
+  }), [formValues]);
+
+  const handleFormChange = (_, { name, value }) => setFormValues(
+    (prev) => ({ ...prev, [name]: value }),
+  );
```

Comment:
This function should calculate the new form values and pass them to `setFormValues`, instead of passing in an intermediary value and then updating that via a side effect. That will be easier to understand, have fewer re-renders, and be less likely to accidentally get into an infinite loop.

Reply:
Okay, does that mean that I should have separate handler for each inputs? (I'm fine with it, but would like to double confirm if I understood it correctly)

Reply:
Oh this is actually nice, useEffect is not needed anymore. Changed it.